src/messaging/ExchangeContext.h

```diff
@@ -126,9 +181,15 @@ class DLL_EXPORT ExchangeContext : public ReferenceCounted<ExchangeContext, Exch
 
     ExchangeDelegate * GetDelegate() const { return mDelegate; }
     void SetDelegate(ExchangeDelegate * delegate) { mDelegate = delegate; }
+    void SetReliableMessageDelegate(ReliableMessageDelegate * delegate) { mReliableMessageContext.SetDelegate(delegate); }
 
     ExchangeManager * GetExchangeMgr() const { return mExchangeMgr; }
 
+    ReliableMessageContext * GetReliableMessageContext() const
+    {
+        return const_cast<ReliableMessageContext *>(&mReliableMessageContext);
+    };
```

Comment:
nit: remove semicolon

why do we allow a const cast here? It sounds as if we should instead reutrn a `const ReliableMessagContext` to be really const correct.

Reply:
ReliableMessageContext needs to be non-const since we need to change its content, such as:
rc->SetAckPending(false);

Remove const keyword to avoid casting.

Reply:
but .... this sounds like a bug. A 'const object' is modified by setting ack pending to false here. This sounds like a logic bug.

Reply:
To be clear, in the PR that was merged `GetReliableMessageContext` is not a const method and the `const_cast` is gone.